src/lib/interfaces/subscription-upgrade-status.ts

```diff
@@ -0,0 +1,22 @@
+export enum SubscriptionUpgradeStatusFlag {
+  PENDING = 0,
+  FAILED = 1,
+  FAILED_CLOSED = 2,
+  SUSPENDED = 3,
+}
```

Comment:
I know the TypeScript community generally frowns upon enums, but I think this implementation has the following benefits:
- preserving the type signature described in the documentation, where the `status` flag is a `number`
- helping the user interpret what the value of `status` means (especially since the values are not documented in Piano's docs)

Reply:
I agree with you. We can see a lot of reasonable usage of TypeScript Enum in many SDKs (e.g. the ones provided by AWS).

I personally prefer String Enums a bit because they look more straightforward even when transpired into JS codes, but I know basic number based enums are appropriate here.

 One problem I face is that I cannot find the meaning of each number  in responses' `status` field in Piano API document (for example, 0 means pending). Where did you know it?

Reply:
@5t111111 
Thanks, glad you agreed about the enum here!

> One problem I face is that I cannot find the meaning of each number in responses' status field in Piano API document (for example, 0 means pending). Where did you know it?

You're right, I learned of those values from a request made to Piano support, and unfortunately I don't think they can be found in any publicly available resource. I'll post this link for reference, though it's from a private repository: https://github.com/kodansha/friday-backlog/issues/998

Reply:
Thank you, would be glad if Piano make that information publicly available. Anyway, I understand the enum values are guaranteed by Piano official!